stactask/task.py

```diff
@@ -138,32 +138,47 @@ def process_definition(self) -> dict[str, Any]:
     @property
     def parameters(self) -> dict[str, Any]:
```

Comment:
This looks great. I'm going to go ahead and approve this and you should probably ignore what follows. :)

I have a couple thoughts about refinements to consider:

* I don't love the name `workflow_config`. I think it was used for consistency with `task_configs`, yeah? I believe that term is only in use because of the deprecated task config list. I would have gone with `workflow_options`, and updated the task verbiage as necessary.
* What about splitting this up into three functions `workflow_options`, `task_options`, and `parameters` (though I don't love the name of this last one I'd keep it for compatibility)? You would return the workflow options loaded from the payload in the first, return the task options loaded from the payload in the second, and return the combined dict of them from the third by calling the first two to get their values.

Reply:
I _really_ like both suggestions. I considered both, but decided to minimize changes. But the fact that you see the same things tells me they should be implemented.

Reply:
Done.